packages/dd-trace/src/appsec/reporter.js

```diff
@@ -140,14 +140,14 @@ function filterExtendedHeaders (headers, excludedHeaderNames, tagPrefix, limit =
   return result
 }
 
-function getCollectedHeaders (req, res, shouldCollectEventHeaders) {
+function getCollectedHeaders (req, res, shouldCollectEventHeaders, storedResponseHeaders) {
   // Mandatory
   const mandatoryCollectedHeaders = filterHeaders(req.headers, REQUEST_HEADERS_MAP)
 
   // Basic collection
   if (!shouldCollectEventHeaders) return mandatoryCollectedHeaders
 
-  const responseHeaders = res.getHeaders()
+  const responseHeaders = { ...storedResponseHeaders, ...res.getHeaders() }
```

Comment:
I know that I suggested this change, but we can't prevent creating copies of the objects when they are not necessary. for example:

```suggestion
  const responseHeaders = storedResponseHeaders ? { ...storedResponseHeaders, ...res.getHeaders() } : res.getHeaders()
```